Classes/DDFileLogger.m

```diff
@@ -869,6 +874,32 @@ - (void)maybeRollLogFileDueToSize {
     }
 }
 
+- (void)maybeRollLogFileDueToDate {
+	if (_rollEveryDay && _currentLogFileInfo && ![self isLogFileCreatedToday:_currentLogFileInfo]) {
+		NSLogVerbose(@"DDFileLogger: Rolling log file due to date...");
+		
+		[self rollLogFileNow];
+	}
+}
+
+- (BOOL)isLogFileCreatedToday:(DDLogFileInfo *)aLogFile {
+	NSDate *logFileCreationDate = aLogFile.creationDate;
+	NSDate *today = [NSDate date];
```

Comment:
I prefer that you modify the file logger so people can easily override the rolling methods. You can make it public and easier to override in a PR.